chain/chain/src/resharding.rs

```diff
@@ -138,26 +162,26 @@ impl Chain {
             new_shard_uid
         };
 
-        let state_root_node = trie.retrieve_root_node()?;
-        let num_parts = get_num_state_parts(state_root_node.memory_usage);
-        debug!(target: "runtime", "splitting state for shard {} to {} parts to build new states", shard_id, num_parts);
-        for part_id in 0..num_parts {
-            let trie_items = trie.get_trie_items_for_part(PartId::new(part_id, num_parts))?;
+        while let Some(batch) = get_trie_update_batch(&mut iter) {
+            // TODO(shreyan): This is highly inefficient as for each key in the batch, we are parsing the account_id
+            // A better way would be to use the boundary account to construct the from and to key range for flat storage iterator
```

Comment:
I've been using TODO(resharding) so that it's easier to find for anyone but up to you, if you're going to be working on it soon it's fine.